DataFormats/METReco/interface/HaloClusterCandidateEE.h

```diff
@@ -0,0 +1,84 @@
+#ifndef DATAFORMATS_METRECO_HALOCLUSTERCANDIDATEEE_H
+#define DATAFORMATS_METRECO_HALOCLUSTERCANDIDATEEE_H
+
+
+
+#include "DataFormats/DetId/interface/DetId.h"
+#include "DataFormats/EcalDetId/interface/EcalSubdetector.h"
+#include "DataFormats/EcalDetId/interface/EBDetId.h"
+#include "DataFormats/EcalDetId/interface/EEDetId.h"
+#include "DataFormats/GeometrySurface/interface/Cylinder.h"
+#include "DataFormats/GeometrySurface/interface/Plane.h"
+#include "DataFormats/GeometrySurface/interface/Cone.h"
+#include "DataFormats/GeometryVector/interface/GlobalPoint.h"
+#include "DataFormats/GeometryVector/interface/GlobalVector.h"
+#include "DataFormats/GeometryVector/interface/LocalPoint.h"
+#include "DataFormats/GeometryVector/interface/LocalVector.h"
+#include "DataFormats/HcalRecHit/interface/HcalRecHitCollections.h"
+#include "DataFormats/EcalRecHit/interface/EcalRecHitCollections.h"
+#include "DataFormats/Math/interface/deltaPhi.h"
+#include "Geometry/CaloGeometry/interface/CaloGeometry.h"
+#include "Geometry/Records/interface/CaloGeometryRecord.h"
+#include "FWCore/Framework/interface/Event.h"
+
+
+namespace reco {
+class HaloClusterCandidateEE {
```

Comment:
... given large overlap with HaloClusterCandidateHB/HE,
the following 
```
  double et;
  double seed_et, seed_eta, seed_phi, seed_Z, seed_R, seed_time;
  bool ishalofrompattern;
  bool ishalofrompattern_hlt;
```
could be packed in a separate class/struct to be reused in all 4 (or 2 if reduced to HCAL and ECAL) classes.